fendermint/app/src/app.rs

```diff
@@ -490,6 +529,47 @@ where
         Ok(response)
     }
 
+    /// 1. Prepare txns
+    /// 2. Prepare parent finality proposal
+    async fn prepare_proposal(
+        &self,
+        request: request::PrepareProposal,
+    ) -> AbciResult<response::PrepareProposal> {
+        let max_tx_bytes: usize = request.max_tx_bytes.try_into().unwrap();
+        let mut size: usize = 0;
+        let mut txs = Vec::new();
+        for tx in request.txs {
+            if size.saturating_add(tx.len()) > max_tx_bytes {
+                break;
+            }
+            size += tx.len();
+            txs.push(tx);
+        }
+
+        if let Some(finality_proposal) = self
+            .parent_finality
+            .next_finality_proposal()
+            .map_err(|e| anyhow!("cannot get next finality to propose due to: {e}"))?
```

Comment:
How can this method fail? It's only reading data from memory, not doing any IO, right? 

Be very careful when you use `?` in `app.rs` because a failure will crash Fendermint and Tendermint will stop as well.